src/net/java/sip/communicator/impl/protocol/jabber/extensions/colibri/ColibriConferenceIQ.java

```diff
@@ -2271,13 +2361,64 @@ public void setDisplayName(String displayName)
         }
 
         /**
+         * Sets the transport element of this <tt>ChannelBundle</tt>.
+         * @param transport the transport to set.
+         */
+        public void setTransport(IceUdpTransportPacketExtension transport)
+        {
+            this.transport = transport;
+        }
+
+        /**
          * Sets the ID of this <tt>Endpoint</tt>.
          * @param id the ID to set.
          */
         public void setId(String id)
         {
             this.id = id;
         }
+
+        /**
+         * Sets the stats ID of this <tt>Endpoint</tt>.
+         * @param statsId the stats ID to set.
+         */
+        public void setStatsId(String statsId)
+        {
+            this.statsId = statsId;
+        }
+
+        /**
+         * Appends the XML <tt>String</tt> representation of this
+         * <tt>Endpoint</tt> to <tt>xml</tt>.
+         *
+         * @param xml the <tt>StringBuilder</tt> to which the XML
+         * <tt>String</tt> representation of this <tt>Endpoint</tt> is to be
+         * appended
+         */
+        public IQChildElementXmlStringBuilder toXML(
+            IQChildElementXmlStringBuilder xml)
+        {
+            xml.halfOpenElement(ELEMENT_NAME)
+                .attribute(ID_ATTR_NAME, id);
+
+            xml.optAttribute(DISPLAYNAME_ATTR_NAME, displayName);
+            xml.optAttribute(STATS_ID_ATTR_NAME, statsId);
+
+
+            IceUdpTransportPacketExtension transport = getTransport();
```

Comment:
You could probably just call addExtension(...) and let Smack deal with adding the extension element xml (and forget about the condition).

Reply:
I don't get this ... I don't see addExtension ... as this is generating the xml ...?

Reply:
See new comments.